### DISCUSSION TITLE
How do I select a font style?

### DISCUSSION THREAD

**Discussion by Hubro** [neovide/neovide#1251] [Q&A] [answered]
I have the font "Source Code Pro" installed on my system with a whole bunch of possible styles like "Light", "Bold", "Regular", "Normal", "Black", "Italic" etc etc.

When I set my font in Neovide:

```
set guifont=Source\ Code\ Pro:h9
```

It uses the "Regular" style by default. How do I specify a different style? I find that "Normal" looks a lot better at 9pt.

**Comment by dineshKumar777** [comment, marked as answer]
Give this a try 

https://github.com/neovide/neovide/issues/955#issuecomment-990871484

**Reply by Hubro** [reply]
Nice! I tested this with `SauceCodePro Nerd Font` and it seems the format is:

```
<font name>:h<font size>:<weight>:<style>
```

Where the weight and style is optional.

The weight can be:

- `w12`: ExtraLight
- `l`: Light
- `(nothing)`: Regular
- `w57`: Medium
- `sb`: SemiBold
- `b`: Bold
- `w87`: Black

And the style can be `i` to enable italics.

(There are certainly other font weights and styles, but these were the ones that my font supported)

So the declaration I'm looking for it:

```
set guifont=SauceCodePro\ Nerd\ Font:h9:w57
```

I tested all these on nvim-qt, and they all work.

I tested all of the above combinations in Neovide, and it seems most of them don't work:

- `w12`: Doesn't work
- `l`: Doesn't work
- `w57`: Doesn't work
- `sb`: Doesn't work
- `b`: Works!
- `w87`: Doesn't work

In fact only "Regular" and "Bold" works, so making it use "Medium" by default seems like it currently isn't possible.

**Reply by Hubro** [reply]
Quick follow-up, I changed color scheme to `ayu` and I discovered that my font is now `w57`, i.e. "Normal", so it seems that the color schemes have control over this.
